src/main/java/org/passay/dictionary/AbstractFileWordList.java

```diff
@@ -299,4 +279,108 @@ private boolean hasRemaining()
       offset = position;
     }
   }
+
+
+  /** Cache of word indices to byte offsets where word starts in backing file. */
+  static class Cache
+  {
+    /** Cache entry that indicates cached word index and byte offset of start of word in backing file. */
+    static class Entry
+    {
+      // CheckStyle:VisibilityModifier OFF
+      /** Cached word index. */
+      int index;
+
+      /** Byte offset where word starts in backing file. */
+      long position;
+      // CheckStyle:VisibilityModifier ON
+
+
+      /**
+       * Creates a new cache entry.
+       *
+       * @param  i  Cached word index.
+       * @param  pos  Byte offset where word starts in backing file.
+       */
+      Entry(final int i, final long pos)
+      {
+        index = i;
+        position = pos;
+      }
+    }
+
+    /** Map of word indices to the byte offset in the file where the word starts. */
+    private LongBuffer map;
+
+    /** Modulus of indices to cache. */
+    private int modulus;
+
+
+    /**
+     * Creates a new cache instance.
+     *
+     * @param  fileSize  Size of file in bytes.
+     * @param  cachePercent  Percent of words to cache.
+     */
+    Cache(final long fileSize, final int cachePercent)
+    {
+      if (cachePercent < 0 || cachePercent > 100) {
+        throw new IllegalArgumentException("cachePercent must be between 0 and 100 inclusive");
+      }
+      final long cacheSize = (fileSize / 100) * cachePercent;
+      if (cacheSize == 0) {
+        return;
+      }
+      modulus = (int) (fileSize / cacheSize);
+
+      if (cacheSize > Integer.MAX_VALUE) {
+        throw new IllegalArgumentException("Cache limit exceeded. Try reducing cacheSize.");
+      }
+      map = ByteBuffer.allocateDirect((int) cacheSize).asLongBuffer();
+    }
+
+
+    /**
+     * Puts an entry that maps the word at given index to the byte offset in into the backing file.
```

Comment:
This should be updated to document the fact that put only succeeds based on the modulus.